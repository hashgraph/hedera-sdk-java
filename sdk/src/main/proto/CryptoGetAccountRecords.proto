syntax = "proto3";

package proto;

/*-
 * ‌
 * Hedera Network Services Protobuf
 * ​
 * Copyright (C) 2018 - 2021 Hedera Hashgraph, LLC
 * ​
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * ‍
 */

option java_package = "com.hedera.hashgraph.sdk.proto";
option java_multiple_files = true;

import "BasicTypes.proto";
import "TransactionRecord.proto";
import "QueryHeader.proto";
import "ResponseHeader.proto";

// Requests records of all transactions for which the given account was the effective payer in the last 3 minutes of consensus time and <tt>ledger.keepRecordsInState=true</tt> was true during <tt>handleTransaction</tt>.
message CryptoGetAccountRecordsQuery {
    QueryHeader header = 1; // Standard info sent from client to node, including the signed payment, and what kind of response is requested (cost, state proof, both, or neither).
    AccountID accountID = 2; // The account ID for which the records should be retrieved
}

// Returns records of all transactions for which the given account was the effective payer in the last 3 minutes of consensus time and <tt>ledger.keepRecordsInState=true</tt> was true during <tt>handleTransaction</tt>.
message CryptoGetAccountRecordsResponse {
    ResponseHeader header = 1; // Standard response from node to client, including the requested fields: cost, or state proof, or both, or neither
    AccountID accountID = 2; // The account that this record is for
    repeated TransactionRecord records = 3; // List of records
}


