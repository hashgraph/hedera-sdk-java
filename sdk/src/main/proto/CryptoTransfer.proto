syntax = "proto3";

package proto;

/*-
 * ‌
 * Hedera Network Services Protobuf
 * ​
 * Copyright (C) 2018 - 2021 Hedera Hashgraph, LLC
 * ​
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * ‍
 */

option java_package = "com.hedera.hashgraph.sdk.proto";
option java_multiple_files = true;

import "BasicTypes.proto";

/* Transfer cryptocurrency from some accounts to other accounts. The accounts list can contain up to 10 accounts. The amounts list must be the same length as the accounts list. Each negative amount is withdrawn from the corresponding account (a sender), and each positive one is added to the corresponding account (a receiver). The amounts list must sum to zero. Each amount is a number of tinyBars (there are 100,000,000 tinyBars in one Hbar).
If any sender account fails to have sufficient hbars, then the entire transaction fails, and none of those transfers occur, though the transaction fee is still charged. This transaction must be signed by the keys for all the sending accounts, and for any receiving accounts that have receiverSigRequired == true. The signatures are in the same order as the accounts, skipping those accounts that don't need a signature.
*/
message CryptoTransferTransactionBody {
    TransferList transfers = 1;
    repeated TokenTransferList tokenTransfers = 2;
}
