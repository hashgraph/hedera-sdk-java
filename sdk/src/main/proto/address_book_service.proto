syntax = "proto3";

package proto;

/*
 * Hedera Network Services Protobuf
 *
 * Copyright (C) 2024 Hedera Hashgraph, LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

option java_package = "com.hedera.hashgraph.sdk.proto";
// <<<pbj.java_package = "com.hedera.hapi.node.addressbook">>> This comment is special code for setting PBJ Compiler java package

import "query.proto";
import "response.proto";
import "transaction_response.proto";
import "transaction.proto";

/**
 * Transactions for the AddressBook Service, those HAPI APIs facilitate changes to the nodes used across the Hedera network.
 * All those transactions needs to be signed by the Hedera Council. Steps needed below:
 * 1. The node operator creates and signs the transaction with their key (the key on the node operator account)
 * 2. The node operator hands this transaction to Alex, who then gives it to the council to sign
 * 3. When signed and submitted, the server will verify that account 2 keys have signed, and the keys on the operator account have signed.
 * Hedera council should have ability to make all edits in addition to add/remove nodes
 */
service AddressBookService {
    /**
      * Prepare to add a new node to the network.
      * When a valid council member initiates a HAPI transaction to add a new node,
      * then the network should acknowledge the transaction and update the networkâ€™s Address Book within 24 hours.
      * The added node will not be active until the network is upgraded.
      */
    rpc createNode (Transaction) returns (TransactionResponse);

    /**
     * Prepare to delete the node to the network.
     * The deleted node will not be deleted until the network is upgraded.
     * Such a deleted node can never be reused.
     */
    rpc deleteNode (Transaction) returns (TransactionResponse);

    /**
     * Prepare to update the node to the network.
     * The node will not be updated until the network is upgraded.
     */
    rpc updateNode (Transaction) returns (TransactionResponse);

    /**
     * Retrieves the node information by node Id.
     */
    rpc getNodeInfo (Query) returns (Response);
}
