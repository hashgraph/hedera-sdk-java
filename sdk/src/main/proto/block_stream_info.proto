/**
 * # Block Stream Info
 * Information stored in consensus state at the beginning of each block to
 * record the status of the immediately prior block.
 *
 * ### Keywords
 * The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT",
 * "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this
 * document are to be interpreted as described in
 * [RFC2119](https://www.ietf.org/rfc/rfc2119) and clarified in
 * [RFC8174](https://www.ietf.org/rfc/rfc8174).
 */
syntax = "proto3";

package com.hedera.hapi.node.state.blockstream;

/*
 * Copyright (C) 2024 Hedera Hashgraph, LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import "timestamp.proto";

option java_package = "com.hedera.hashgraph.sdk.proto";
// <<<pbj.java_package = "com.hedera.hapi.node.state.blockstream">>> This comment is special code for setting PBJ Compiler java package
option java_multiple_files = true;

/**
 * A message stored in state to maintain block stream parameters.<br/>
 * Nodes use this information for three purposes.
 * 1. To maintain hash chain continuity at restart and reconnect boundaries.
 * 1. To store historical hashes for implementation of the EVM `BLOCKHASH`
 *    and `PREVRANDAO` opcodes.
 * 1. To track the amount of consensus time that has passed between blocks.
 *
 * This value MUST be updated for every block.<br/>
 * This value MUST be transmitted in the "state changes" section of
 * _each_ block, but MUST be updated at the beginning of the _next_ block.<br/>
 * This value SHALL contain the block hash up to, and including, the
 * immediately prior completed block.
 */
message BlockStreamInfo {
    /**
     * A block number.<br/>
     * This is the current block number.
     */
    uint64 block_number = 1;

    /**
     * A consensus time for the current block.<br/>
     * This is the _first_ consensus time in the current block, and
     * is used to determine if this block was the first across an
     * important boundary in consensus time, such as UTC midnight.
     * This may also be used to purge entities expiring between the last
     * block time and this time.
     */
    proto.Timestamp block_time = 2;

    /**
     * A concatenation of hash values.<br/>
     * This combines several trailing output block item hashes and
     * is used as a seed value for a pseudo-random number generator.<br/>
     * This is also requiried to implement the EVM `PREVRANDAO` opcode.
     */
    bytes trailing_output_hashes = 3;

    /**
     * A concatenation of hash values.<br/>
     * This field combines up to 256 trailing block hashes.
     * <p>
     * If this message is for block number N, then the earliest available
     * hash SHALL be for block number N-256.<br/>
     * The latest available hash SHALL be for block N-1.<br/>
     * This is REQUIRED to implement the EVM `BLOCKHASH` opcode.
     */
    bytes trailing_block_hashes = 4;
}
