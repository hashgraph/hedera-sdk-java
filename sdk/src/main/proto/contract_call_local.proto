syntax = "proto3";

package proto;

/*-
 * ‌
 * Hedera Network Services Protobuf
 * ​
 * Copyright (C) 2018 - 2021 Hedera Hashgraph, LLC
 * ​
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * ‍
 */

option java_package = "com.hedera.hashgraph.sdk.proto";
option java_multiple_files = true;

import "basic_types.proto";
import "query_header.proto";
import "response_header.proto";

/* The log information for an event returned by a smart contract function call. One function call may return several such events. */
message ContractLoginfo {
    ContractID contractID = 1; // address of a contract that emitted the event
    bytes bloom = 2; // bloom filter for a particular log
    repeated bytes topic = 3; // topics of a particular event
    bytes data = 4; // event data
}

/* The result returned by a call to a smart contract function. This is part of the response to a ContractCallLocal query, and is in the record for a ContractCall or ContractCreateInstance transaction. The ContractCreateInstance transaction record has the results of the call to the constructor. */
message ContractFunctionResult {
    ContractID contractID = 1; // the smart contract instance whose function was called
    bytes contractCallResult = 2; // the result returned by the function
    string errorMessage = 3; // message In case there was an error during smart contract execution
    bytes bloom = 4; // bloom filter for record
    uint64 gasUsed = 5; // units of gas used to execute contract
    repeated ContractLoginfo logInfo = 6; // the log info for events returned by the function
    repeated ContractID createdContractIDs = 7; // the list of smart contracts that were created by the function call
}

/*
Call a function of the given smart contract instance, giving it functionParameters as its inputs.
This is performed locally on the particular node that the client is communicating with.
It cannot change the state of the contract instance (and so, cannot spend anything from the instance's cryptocurrency account).
It will not have a consensus timestamp. It cannot generate a record or a receipt. The response will contain the output
returned by the function call.  This is useful for calling getter functions, which purely read the state and don't change it.
It is faster and cheaper than a normal call, because it is purely local to a single  node.

Unlike a ContractCall transaction, the node will consume the entire amount of provided gas in determining
the fee for this query.
*/
message ContractCallLocalQuery {
    QueryHeader header = 1; // standard info sent from client to node, including the signed payment, and what kind of response is requested (cost, state proof, both, or neither). The payment must cover the fees and all of the gas offered.
    ContractID contractID = 2; // the contract instance to call, in the format used in transactions
    int64 gas = 3; // The amount of gas to use for the call; all of the gas offered will be used and charged a corresponding fee
    bytes functionParameters = 4; // which function to call, and the parameters to pass to the function
    int64 maxResultSize = 5; // max number of bytes that the result might include. The run will fail if it would have returned more than this number of bytes.
}

/* Response when the client sends the node ContractCallLocalQuery */
message ContractCallLocalResponse {
    ResponseHeader header = 1; //standard response from node to client, including the requested fields: cost, or state proof, or both, or neither
    ContractFunctionResult functionResult = 2; // the value returned by the function (if it completed and didn't fail)
}
