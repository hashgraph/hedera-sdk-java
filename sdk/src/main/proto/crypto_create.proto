syntax = "proto3";

package proto;

/*-
 * ‌
 * Hedera Network Services Protobuf
 * ​
 * Copyright (C) 2018 - 2021 Hedera Hashgraph, LLC
 * ​
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * ‍
 */

option java_package = "com.hedera.hashgraph.sdk.proto";
option java_multiple_files = true;

import "basic_types.proto";
import "duration.proto";

/*
 * Create a new account. After the account is created, the AccountID for it is in the receipt. It
 * can also be retrieved with a GetByKey query. Threshold values can be defined, and records are
 * generated and stored for 25 hours for any transfer that exceeds the thresholds. This account is
 * charged for each record generated, so the thresholds are useful for limiting record generation to
 * happen only for large transactions.
 *
 * The Key field is the key used to sign transactions for this account. If the account has
 * receiverSigRequired set to true, then all cryptocurrency transfers must be signed by this
 * account's key, both for transfers in and out. If it is false, then only transfers out have to be
 * signed by it. When the account is created, the payer account is charged enough hbars so that the
 * new account will not expire for the next autoRenewPeriod seconds. When it reaches the expiration
 * time, the new account will then be automatically charged to renew for another autoRenewPeriod
 * seconds. If it does not have enough hbars to renew for that long, then the remaining hbars are
 * used to extend its expiration as long as possible. If it is has a zero balance when it expires,
 * then it is deleted. This transaction must be signed by the payer account. If receiverSigRequired
 * is false, then the transaction does not have to be signed by the keys in the keys field. If it is
 * true, then it must be signed by them, in addition to the keys of the payer account. If the 
 * auto_renew_account field is set, the key of the referenced account must sign.
 *
 * An entity (account, file, or smart contract instance) must be created in a particular realm. If
 * the realmID is left null, then a new realm will be created with the given admin key. If a new
 * realm has a null adminKey, then anyone can create/modify/delete entities in that realm. But if an
 * admin key is given, then any transaction to create/modify/delete an entity in that realm must be
 * signed by that key, though anyone can still call functions on smart contract instances that exist
 * in that realm. A realm ceases to exist when everything within it has expired and no longer
 * exists.
 *
 * The current API ignores shardID, realmID, and newRealmAdminKey, and creates everything in shard 0
 * and realm 0, with a null key. Future versions of the API will support multiple realms and
 * multiple shards.
 */
message CryptoCreateTransactionBody {
    /**
     * The key that must sign each transfer out of the account. If receiverSigRequired is true, then
     * it must also sign any transfer into the account.
     */
    Key key = 1;

    /**
     * The initial number of tinybars to put into the account
     */
    uint64 initialBalance = 2;

    /**
     * [Deprecated] ID of the account to which this account is proxy staked. If proxyAccountID is null, or is an
     * invalid account, or is an account that isn't a node, then this account is automatically proxy
     * staked to a node chosen by the network, but without earning payments. If the proxyAccountID
     * account refuses to accept proxy staking , or if it is not currently running a node, then it
     * will behave as if proxyAccountID was null.
     */
    AccountID proxyAccountID = 3 [deprecated = true];

    /**
     * [Deprecated]. The threshold amount (in tinybars) for which an account record is created for
     * any send/withdraw transaction
     */
    uint64 sendRecordThreshold = 6 [deprecated=true];

    /**
     * [Deprecated]. The threshold amount (in tinybars) for which an account record is created for
     * any receive/deposit transaction
     */
    uint64 receiveRecordThreshold = 7 [deprecated=true];

    /**
     * If true, this account's key must sign any transaction depositing into this account (in
     * addition to all withdrawals)
     */
    bool receiverSigRequired = 8;

    /**
     * The account is charged to extend its expiration date every this many seconds. If it doesn't
     * have enough balance, it extends as long as possible. If it is empty when it expires, then it
     * is deleted.
     */
    Duration autoRenewPeriod = 9;

    /**
     * The shard in which this account is created
     */
    ShardID shardID = 10;

    /**
     * The realm in which this account is created (leave this null to create a new realm)
     */
    RealmID realmID = 11;

    /**
     * If realmID is null, then this the admin key for the new realm that will be created
     */
    Key newRealmAdminKey = 12;

    /**
     * The memo associated with the account (UTF-8 encoding max 100 bytes)
     */
    string memo = 13;

    /**
     * The maximum number of tokens that an Account can be implicitly associated with. Defaults to 0
     * and up to a maximum value of 1000.
     */
    int32 max_automatic_token_associations = 14;

    /**
     * ID of the account or node to which this account is staking.
     */
    oneof staked_id {

        /**
         * ID of the account to which this account is staking.
         */
        AccountID staked_account_id = 15;

        /**
         * ID of the node this account is staked to.
         */
        int64 staked_node_id = 16;
    }

    /**
     * If true, the account declines receiving a staking reward. The default value is false.
     */
    bool decline_reward = 17;

    /**
     * The bytes to be used as the account's alias. The bytes will be 1 of 2 options. It will be the 
     * serialization of a protobuf Key message for an ED25519/ECDSA_SECP256K1 primitive key type. If the 
     * account is ECDSA_SECP256K1 based it may also be the public address, calcluated as the last 20 bytes of
     * the keccak-256 hash of the ECDSA_SECP256K1 primitive key. Currently only primitive key bytes are 
     * supported as the key for an account with an alias. ThresholdKey, KeyList, ContractID, and
     * delegatable_contract_id are not supported.
     *
     * A given alias can map to at most one account on the network at a time. This uniqueness will be enforced
     * relative to aliases currently on the network at alias assignment.
     *
     * If a transaction creates an account using an alias, any further crypto transfers to that alias will 
     * simply be deposited in that account, without creating anything, and with no creation fee being charged.
     */
     bytes alias = 18;

    /**
     * An account to charge for auto-renewal of this account . If not set, or set to an
     * account with zero hbar balance, the account's own hbar balance will be used to
     * cover auto-renewal fees.
     */
    AccountID auto_renew_account = 19;
}
