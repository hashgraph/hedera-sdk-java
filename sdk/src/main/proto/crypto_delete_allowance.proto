syntax = "proto3";

package proto;

/*-
 * ‌
 * Hedera Network Services Protobuf
 * ​
 * Copyright (C) 2018 - 2022 Hedera Hashgraph, LLC
 * ​
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * ‍
 */

option java_package = "com.hedera.hashgraph.sdk.proto";
option java_multiple_files = true;

import "basic_types.proto";

/**
 * Deletes one or more hbar/token approved allowances from an owner's account. This operation
 * will remove the allowances granted to <b>all</b> spenders of an owner's hbar, a specific
 * fungible token and/or a specific non-fungible token serial number. Each owner account
 * listed as wiping an allowance must sign the transaction.
 */
message CryptoDeleteAllowanceTransactionBody {
    /**
     * List of hbar allowances to remove.
     */
    repeated CryptoRemoveAllowance cryptoAllowances = 1;

    /**
     * List of non-fungible token allowances to remove.
     */
    repeated NftRemoveAllowance nftAllowances = 2;

    /**
     * List of fungible token allowances to remove.
     */
    repeated TokenRemoveAllowance tokenAllowances = 3;
}

/**
 * Hbar allowances to be removed on an owner account
 */
message CryptoRemoveAllowance {
    /**
     * The account ID of the hbar owner (ie. the grantor of the allowance).
     */
    AccountID owner = 1;
}

/**
 * Nft allowances to be removed on an owner account
 */
message NftRemoveAllowance {
    /**
     * The token that the allowance pertains to.
     */
    TokenID token_id = 1;

    /**
     * The account ID of the token owner (ie. the grantor of the allowance).
     */
    AccountID owner = 2;

    /**
     * The list of serial numbers to remove allowances from.
     */
    repeated int64 serial_numbers = 3;
}

/**
 * Token allowances to be removed on an owner account
 */
message TokenRemoveAllowance {
    /**
     * The token that the allowance pertains to.
     */
    TokenID token_id = 1;

    /**
     * The account ID of the token owner (ie. the grantor of the allowance).
     */
    AccountID owner = 2;
}
