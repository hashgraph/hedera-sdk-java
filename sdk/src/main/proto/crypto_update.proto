syntax = "proto3";

package proto;

/*-
 * ‌
 * Hedera Network Services Protobuf
 * ​
 * Copyright (C) 2018 - 2021 Hedera Hashgraph, LLC
 * ​
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * ‍
 */

option java_package = "com.hedera.hashgraph.sdk.proto";
option java_multiple_files = true;

import "basic_types.proto";
import "duration.proto";
import "timestamp.proto";
import "google/protobuf/wrappers.proto";

/*
Change properties for the given account. Any null field is ignored (left unchanged). This transaction must be signed by the existing key for this account. If the transaction is changing the key field, then the transaction must be signed by both the old key (from before the change) and the new key. The old key must sign for security. The new key must sign as a safeguard to avoid accidentally changing to an invalid key, and then having no way to recover.
*/
message CryptoUpdateTransactionBody {
    AccountID accountIDToUpdate = 2; // The account ID which is being updated in this transaction
    Key key = 3; // The new key
    AccountID proxyAccountID = 4; // ID of the account to which this account is proxy staked. If proxyAccountID is null, or is an invalid account, or is an account that isn't a node, then this account is automatically proxy staked to a node chosen by the network, but without earning payments. If the proxyAccountID account refuses to accept proxy staking , or if it is not currently running a node, then it will behave as if proxyAccountID was null.
    int32 proxyFraction = 5 [deprecated = true]; // [Deprecated]. Payments earned from proxy staking are shared between the node and this account, with proxyFraction / 10000 going to this account
    oneof sendRecordThresholdField {
        uint64 sendRecordThreshold = 6 [deprecated = true]; // [Deprecated]. The new threshold amount (in tinybars) for which an account record is created for any send/withdraw transaction
        google.protobuf.UInt64Value sendRecordThresholdWrapper = 11 [deprecated = true]; // [Deprecated]. The new threshold amount (in tinybars) for which an account record is created for any send/withdraw transaction
    }
    oneof receiveRecordThresholdField {
        uint64 receiveRecordThreshold = 7 [deprecated = true]; // [Deprecated]. The new threshold amount (in tinybars) for which an account record is created for any receive/deposit transaction.
        google.protobuf.UInt64Value receiveRecordThresholdWrapper = 12 [deprecated = true]; // [Deprecated]. The new threshold amount (in tinybars) for which an account record is created for any receive/deposit transaction.
    }
    Duration autoRenewPeriod = 8; // The duration in which it will automatically extend the expiration period. If it doesn't have enough balance, it extends as long as possible. If it is empty when it expires, then it is deleted.
    Timestamp expirationTime = 9; // The new expiration time to extend to (ignored if equal to or before the current one)
    oneof receiverSigRequiredField {
        bool receiverSigRequired = 10 [deprecated = true];  // [Deprecated] Do NOT use this field to set a false value because the server cannot distinguish from the default value. Use receiverSigRequiredWrapper field for this purpose.
        google.protobuf.BoolValue receiverSigRequiredWrapper = 13; // If true, this account's key must sign any transaction depositing into this account (in addition to all withdrawals)
    }
    google.protobuf.StringValue memo = 14; // If set, the new memo to be associated with the account (UTF-8 encoding max 100 bytes)
    google.protobuf.UInt32Value max_automatic_token_associations = 15; // The maximum number of tokens that an Account can be implicitly associated with. Up to a 1000 including implicit and explicit associations.
}
