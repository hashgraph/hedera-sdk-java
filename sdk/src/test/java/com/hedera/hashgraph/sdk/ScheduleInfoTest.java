package com.hedera.hashgraph.sdk;

import com.hedera.hashgraph.sdk.proto.CryptoDeleteTransactionBody;
import com.hedera.hashgraph.sdk.proto.SchedulableTransactionBody;
import io.github.jsonSnapshot.SnapshotMatcher;
import org.junit.AfterClass;
import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.Test;
import java.time.Instant;

import static org.assertj.core.api.Assertions.assertThat;

public class ScheduleInfoTest {
    private static final PublicKey unusedPublicKey = PrivateKey.fromString(
        "302e020100300506032b657004220420db484b828e64b2d8f12ce3c0a0e93a0b8cce7af1bb8f39c97732394482538e10"
    ).getPublicKey();

    final Instant validStart = Instant.ofEpochSecond(1554158542);

    @BeforeAll
    public static void beforeAll() {
        SnapshotMatcher.start();
    }

    @AfterClass
    public static void afterAll() {
        SnapshotMatcher.validateSnapshots();
    }

    ScheduleInfo spawnScheduleInfoExample() {
        return new ScheduleInfo(
            ScheduleId.fromString("1.2.3"),
            AccountId.fromString("4.5.6"),
            AccountId.fromString("2.3.4"),
            SchedulableTransactionBody.newBuilder()
                .setCryptoDelete(CryptoDeleteTransactionBody.newBuilder()
                    .setDeleteAccountID(AccountId.fromString("6.6.6").toProtobuf()).build()).build(),
            KeyList.of(unusedPublicKey),
            unusedPublicKey,
            TransactionId.withValidStart(AccountId.fromString("0.0.5006"), validStart),
            "memo",
            validStart,
            validStart,
            null,
            LedgerId.TESTNET,
            true
        );
    }

    ScheduleInfo spawnScheduleInfoDeletedExample() {
        return new ScheduleInfo(
            ScheduleId.fromString("1.2.3"),
            AccountId.fromString("4.5.6"),
            AccountId.fromString("2.3.4"),
            SchedulableTransactionBody.newBuilder()
                .setCryptoDelete(CryptoDeleteTransactionBody.newBuilder()
                    .setDeleteAccountID(AccountId.fromString("6.6.6").toProtobuf()).build()).build(),
            KeyList.of(unusedPublicKey),
            unusedPublicKey,
            TransactionId.withValidStart(AccountId.fromString("0.0.5006"), validStart),
            "memo",
            validStart,
            null,
            validStart,
            LedgerId.TESTNET,
            true
        );
    }

    @Test
    void shouldSerialize() throws Exception {
        var originalScheduleInfo = spawnScheduleInfoExample();
        byte[] scheduleInfoBytes = originalScheduleInfo.toBytes();
        var copyScheduleInfo = ScheduleInfo.fromBytes(scheduleInfoBytes);
        assertThat(copyScheduleInfo.toString().replaceAll("@[A-Za-z0-9]+", ""))
            .isEqualTo(originalScheduleInfo.toString().replaceAll("@[A-Za-z0-9]+", ""));
        SnapshotMatcher.expect(originalScheduleInfo.toString().replaceAll("@[A-Za-z0-9]+", "")).toMatchSnapshot();
    }

    @Test
    void shouldSerializeDeleted() throws Exception {
        var originalScheduleInfo = spawnScheduleInfoDeletedExample();
        byte[] scheduleInfoBytes = originalScheduleInfo.toBytes();
        var copyScheduleInfo = ScheduleInfo.fromBytes(scheduleInfoBytes);
        assertThat(copyScheduleInfo.toString().replaceAll("@[A-Za-z0-9]+", ""))
            .isEqualTo(originalScheduleInfo.toString().replaceAll("@[A-Za-z0-9]+", ""));
        SnapshotMatcher.expect(originalScheduleInfo.toString().replaceAll("@[A-Za-z0-9]+", "")).toMatchSnapshot();
    }
}
