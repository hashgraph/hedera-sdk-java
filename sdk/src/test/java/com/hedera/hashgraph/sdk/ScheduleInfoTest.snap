com.hedera.hashgraph.sdk.ScheduleInfoTest.shouldSerialize=[
  "ScheduleInfo{scheduleId=1.2.3, scheduledTransactionId=0.0.5006.000000000, creatorAccountId=4.5.6, payerAccountId=2.3.4, signatories=KeyList{threshold=null, keys=[302a300506032b6570032100e0c8ec2758a5879ffac226a13c0c516b799e72e35141a0dd828f94d37988a4b7]}, adminKey=302a300506032b6570032100e0c8ec2758a5879ffac226a13c0c516b799e72e35141a0dd828f94d37988a4b7, expirationTime=2019-04-01T22:42:22Z, memo=memo, executedAt=2019-04-01T22:42:22Z, deletedAt=null, ledgerId=testnet, waitForExpiry=true}"
]


com.hedera.hashgraph.sdk.ScheduleInfoTest.shouldSerializeDeleted=[
  "ScheduleInfo{scheduleId=1.2.3, scheduledTransactionId=0.0.5006.000000000, creatorAccountId=4.5.6, payerAccountId=2.3.4, signatories=KeyList{threshold=null, keys=[302a300506032b6570032100e0c8ec2758a5879ffac226a13c0c516b799e72e35141a0dd828f94d37988a4b7]}, adminKey=302a300506032b6570032100e0c8ec2758a5879ffac226a13c0c516b799e72e35141a0dd828f94d37988a4b7, expirationTime=2019-04-01T22:42:22Z, memo=memo, executedAt=null, deletedAt=2019-04-01T22:42:22Z, ledgerId=testnet, waitForExpiry=true}"
]
